main/Main.hs

```diff
@@ -324,11 +332,12 @@ loadDbUriFile conf = extractDbUri mDbUri
         Just filename -> strip <$> readFile (toS filename)
     setDbUri dbUri = conf {configDatabase = dbUri}
 
-createAndBindSocket :: FilePath -> IO Socket
-createAndBindSocket filePath = do
-  deleteSocketFileIfExist filePath
+createAndBindSocket :: FilePath -> FileMode -> IO Socket
+createAndBindSocket socketFilePath socketFileMode = do
+  deleteSocketFileIfExist socketFilePath
   sock <- socket AF_UNIX Stream defaultProtocol
-  bind sock $ SockAddrUnix filePath
+  bind sock $ SockAddrUnix socketFilePath
+  setFileMode socketFilePath socketFileMode
```

Comment:
How about if we just leave it with the default of `755`(also no breaking change) and let the `setFileMode` be optional:

```haskell
when (isJust socketFileMode) $
  setFileMode socketFilePath $ fromJust socketFileMode
```

Then you could call this function like: `createAndBindSocket (unpack socketAddr) (rightToMaybe socketFileMode)`.

This way you could delete the `432` parts and the validation would be similar for all configs options in the `main` func.

Reply:
@steve-chavez thank you for your feedback, the suggestion made a lot of sense.

Reply:
FYI I ended up changing it to forM_ to comply with linters and then again to mapM_ for better readability

Reply:
Oh, that's better. Lately I've been noticing that there are parts of the code that could be simplified in terms of `Traversable`: 
```hs
setFileMode socketFilePath `traverse_` maybeSocketFileMode
```

But in this case I think the `mapM_` is more readable, so it's good to leave it as it is.

I'll give the PR another review.